precompiles/balances-erc20/src/mock.rs

```diff
@@ -0,0 +1,278 @@
+use frame_support::{
+	construct_runtime, parameter_types,
+	traits::{Everything, OnFinalize, OnInitialize},
+};
+use pallet_evm::{AddressMapping, EnsureAddressNever, EnsureAddressRoot, PrecompileSet};
+use serde::{Deserialize, Serialize};
+use sp_core::{H160, H256};
+use sp_runtime::{
+	testing::Header,
+	traits::{BlakeTwo256, IdentityLookup},
+};
+
+pub const PRECOMPILE_ADDRESS: u64 = 1;
+
+pub type AccountId = Account;
+pub type Balance = u128;
+pub type BlockNumber = u64;
+pub type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<Runtime>;
+pub type Block = frame_system::mocking::MockBlock<Runtime>;
+
+/// A simple account type.
+#[derive(
+	Eq,
+	PartialEq,
+	Ord,
+	PartialOrd,
+	Clone,
+	Encode,
+	Decode,
+	Debug,
+	MaxEncodedLen,
+	Serialize,
+	Deserialize,
+	derive_more::Display,
+)]
+pub enum Account {
+	Alice,
+	Bob,
+	Charlie,
+	Bogus,
+	Precompile,
+}
+
+impl Default for Account {
+	fn default() -> Self {
+		Self::Bogus
+	}
+}
+
+impl AddressMapping<Account> for Account {
+	fn into_account_id(h160_account: H160) -> Account {
+		match h160_account {
+			a if a == H160::repeat_byte(0xAA) => Self::Alice,
+			a if a == H160::repeat_byte(0xBB) => Self::Bob,
+			a if a == H160::repeat_byte(0xCC) => Self::Charlie,
+			a if a == H160::from_low_u64_be(PRECOMPILE_ADDRESS) => Self::Precompile,
+			_ => Self::Bogus,
+		}
+	}
+}
+
+impl From<Account> for H160 {
+	fn from(x: Account) -> H160 {
+		match x {
+			Account::Alice => H160::repeat_byte(0xAA),
+			Account::Bob => H160::repeat_byte(0xBB),
+			Account::Charlie => H160::repeat_byte(0xCC),
+			Account::Precompile => H160::from_low_u64_be(PRECOMPILE_ADDRESS),
+			Account::Bogus => Default::default(),
+		}
+	}
+}
+
+impl From<H160> for Account {
+	fn from(x: H160) -> Account {
+		Account::into_account_id(x)
+	}
+}
+
+impl From<Account> for H256 {
+	fn from(x: Account) -> H256 {
+		let x: H160 = x.into();
+		x.into()
+	}
+}
+
+parameter_types! {
+	pub const BlockHashCount: u64 = 250;
+	pub const SS58Prefix: u8 = 42;
+}
+
+impl frame_system::Config for Runtime {
+	type BaseCallFilter = Everything;
+	type DbWeight = ();
+	type Origin = Origin;
+	type Index = u64;
+	type BlockNumber = BlockNumber;
+	type Call = Call;
+	type Hash = H256;
+	type Hashing = BlakeTwo256;
+	type AccountId = AccountId;
+	type Lookup = IdentityLookup<Self::AccountId>;
+	type Header = Header;
+	type Event = Event;
+	type BlockHashCount = BlockHashCount;
+	type Version = ();
+	type PalletInfo = PalletInfo;
+	type AccountData = pallet_balances::AccountData<Balance>;
+	type OnNewAccount = ();
+	type OnKilledAccount = ();
+	type SystemWeightInfo = ();
+	type BlockWeights = ();
+	type BlockLength = ();
+	type SS58Prefix = SS58Prefix;
+	type OnSetCode = ();
+}
+
+parameter_types! {
+	pub const MinimumPeriod: u64 = 5;
+}
+
+impl pallet_timestamp::Config for Runtime {
+	type Moment = u64;
+	type OnTimestampSet = ();
+	type MinimumPeriod = MinimumPeriod;
+	type WeightInfo = ();
+}
```

Comment:
Is pallet timestamp used at all in your mock or testing?

Reply:
It is not, I just copied from the other mock. If you think it can easily be removed I'll do it :)

Reply:
pallet_evm directly depends on it last I check, so it does have to be here